pkg/api/api_implementation.go

```diff
@@ -92,7 +93,27 @@ func (a defaultAPI) Create(ctx context.Context, o types.Object, opts ...types.Cr
 		opt.ApplyToCreate(&options)
 	}
 
-	return a.do(ctx, o, o, &options, types.OperationCreate)
+	if err := a.do(ctx, o, o, &options, types.OperationCreate); err != nil {
+		return fmt.Errorf("failed to execute request: %w", err)
+	}
+
+	if err := a.handlePostCreateOptions(ctx, o, options); err != nil {
+		return fmt.Errorf("failed running post create options: %w", err)
```

Comment:
let's not wrap the error here with another message, instead return a good message from `a.handlePostCreateOptions()` (you already return a good message)